lib/3scale_toolbox/tasks/copy_pricingrules_task.rb

```diff
@@ -0,0 +1,48 @@
+module ThreeScaleToolbox
+  module Tasks
+    class CopyPricingRulesTask
+      include CopyTask
+      include Helper
+
+      def call
+        metrics_map = metrics_mapping(source.metrics, target.metrics)
+        plan_mapping = application_plan_mapping(source.plans, target.plans)
+        plan_mapping.each do |plan_id, target_plan|
+          pricingrules_source = source.remote.list_pricingrules_per_application_plan(plan_id)
+          pricingrules_target = target.remote.list_pricingrules_per_application_plan(target_plan['id'])
+          missing_pricingrules = missing_pricingrules(pricingrules_source, pricingrules_target)
+          missing_pricingrules.each do |pricingrule|
+            links = pricingrule.delete('links')
+            target.remote.create_pricingrule(
+              target_plan['id'],
+              metrics_map.fetch(parse_metric_id_from_links(links)),
+              pricingrule
+            )
+          end
+          puts "Missing #{missing_pricingrules.size} pricing rules from target application plan " \
+            "#{target_plan['id']}. Source plan #{plan_id}"
+        end
+      end
+
+      private
+
+      def missing_pricingrules(source_pricingrules, target_pricingrules)
+        ThreeScaleToolbox::Helper.array_difference(source_pricingrules, target_pricingrules) do |src, target|
+          ThreeScaleToolbox::Helper.compare_hashes(src, target, ['system_name'])
+        end
+      end
+
+      def parse_metric_id_from_links(pricing_rule_links)
+        # This method will not be necessary when https://github.com/3scale/porta/issues/671 is fixed.
+        # When method
+        # https://3scale-supertest-admin.3scale.net/admin/api/services/2555417769646/metrics/2555418155744/methods/2555418155754
+        # When metric
+        # https://3scale-supertest-admin.3scale.net/admin/api/services/2555417769646/metrics/2555418160039
+        metric_link = pricing_rule_links.select { |link| link['rel'] == 'metric' }[0]
```

Comment:
Might be worth waiting for https://github.com/3scale/porta/pull/688 ?

Reply:
we will wait for it